src/include/duckdb/planner/operator/logical_insert.hpp

```diff
@@ -30,8 +35,19 @@ class LogicalInsert : public LogicalOperator {
 	vector<unique_ptr<Expression>> bound_defaults;
 
 protected:
+	vector<ColumnBinding> GetColumnBindings() override {
+		if (return_chunk) {
+			return GenerateColumnBindings(table_index, table->columns.size());
+		}
+		return {ColumnBinding(0, 0)};
+	}
+
 	void ResolveTypes() override {
-		types.emplace_back(LogicalType::BIGINT);
+		if (return_chunk) {
+			types = table->GetTypes();
```

Comment:
This is incorrect, no? The types returned here should be the types returned by the `RETURNING` expression, not the types of the table (which can be different or in a different order).